src/Pages/Productpage/Productpage.jsx

```diff
@@ -1,6 +1,120 @@
+import { Filter } from "../../Components/Productpage/";
+import { useFilter } from "../../CustomHooks/useFilter";
+import { useState, useEffect } from "react";
+import { useFetch } from "../../CustomHooks/useFetch";
+import axios from "axios";
+import ProductList from "../../Components/Productpage/ProductListing/ProductList";
+import spinner from "../../Assets/loader";
+import "./Productpage.css";
+
 const Productpage = () => {
-    return <div className="sub-container">Productpage</div>;
 
-}
 
-export default Productpage;
\ No newline at end of file
+  const [products, setProducts] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const { filterState, filterDispatch } = useFilter();
+
+   // fetch products
+  // const { data, error, loading } = useFetch({
+  //   url: "/api/products",
+  //   method: "GET",
+  // });
+
+  // filterDispatch({ type: "SET_PRODUCTS", payload: data?.products });
+
+  useEffect(() => {
+   const fetchData = async () => {
+      try {
+        const response = await axios({
+          url: "/api/products",
+          method: "GET",
+        });
+        if (response.status === 200 || response.status === 201) {
+          setProducts(response.data.products);
+          filterDispatch({ type: "SET_ITEMS", payload: response.data.products});
+          setLoading(false);
+        } else {
+          setError(response.statusText);
+        }
```

Comment:
You can remove the else block as the code will go directly to the catch block if there is any other status code.